application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,165 @@
+                "What do you want to tell them?", true);
+        OptionData guildOption = new OptionData(OptionType.STRING, OPTION_SERVER_GUILD,
+                "Your server guild ID", true);
+        OptionData anonymousOption = new OptionData(OptionType.BOOLEAN, OPTION_STAY_ANONYMOUS,
+                "if set, your name is hidden", true);
+
+        List<Command.Choice> choices = jda.getGuildCache()
+            .stream()
+            .map(guild -> new Command.Choice(guild.getName(), guild.getIdLong()))
+            .toList();
+
+        guildOption.addChoices(choices);
+
+        getData().addOptions(messageOption, guildOption, anonymousOption);
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        long userId = event.getUser().getIdLong();
+        event.deferReply().setEphemeral(true).queue();
+
+        if (isChannelOnCooldown(userId)) {
+            event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+        authorIdToLastCommandInvocation.put(userId, Instant.now());
+
+        Optional<TextChannel> modMailAuditLog = getChannel(event);
+        if (modMailAuditLog.isEmpty()) {
+            logger.warn("Cannot find the designated modmail channel in this server.");
+            return;
+        }
+
+        String userMessage = event.getOption(OPTION_MESSAGE).getAsString();
+        boolean wantsToStayAnonymous = event.getOption(OPTION_STAY_ANONYMOUS).getAsBoolean();
+        String user = event.getUser().getAsMention();
+        MessageCreateAction message = modMailAuditLog.orElseThrow()
+            .sendMessageEmbeds(createModMailMessage(user, userMessage))
+            .addActionRow(DiscordClientAction.General.USER.asLinkButton("Click to see profile!",
+                    String.valueOf(userId)));
+
+        if (wantsToStayAnonymous) {
+            user = "Anonymous";
+            message = modMailAuditLog.orElseThrow()
+                .sendMessageEmbeds(createModMailMessage(user, userMessage));
+        }
+
+        message.mapToResult().flatMap(result -> {
+            if (result.isSuccess()) {
+                return event.getHook().editOriginal("Your message has been forwarded, thanks.");
+            } else {
+                return event.getHook()
+                    .editOriginal("There was an issue forwarding your message, sorry. We are "
+                            + "investigating.");
+            }
```

Comment:
duplication on `return event.getHook().editOriginal`. Instead, do:
```java
.map(result -> result.isSuccess() ? "Your message ..." : "There was an issue ...")
.flatMap(message -> event.getHook().editOriginal(message))
```

Also, you are not logging the issue at all, how should we investigate it then? You need to log the issue, obtainable in `result`.